File: vision_solo.py
import os
import json
import base64
import requests
import sys

# Path to Claude Code credentials
CRED_PATH = os.path.expanduser("~/.claude/.credentials.json")

def get_token():
    try:
        if not os.path.exists(CRED_PATH):
            print(f"Error: Credentials file not found at {CRED_PATH}")
            return None
        with open(CRED_PATH, 'r') as f:
            creds = json.load(f)
            return creds.get("claudeAiOauth", {}).get("accessToken")
    except Exception as e:
        print(f"Error reading credentials: {e}")
        return None

def analyze_image(image_path, prompt="Analyze this image."):
    token = get_token()
    if not token:
        print("Error: Could not find Claude Code access token. Please login using 'claude login'.")
        return

    try:
        with open(image_path, "rb") as image_file:
            data = image_file.read()
            encoded_string = base64.b64encode(data).decode('utf-8')
            
            # Simple mime-type detection based on extension
            ext = os.path.splitext(image_path)[1].lower()
            mime_type_map = {
                '.jpg': 'image/jpeg',
                '.jpeg': 'image/jpeg',
                '.png': 'image/png',
                '.gif': 'image/gif',
                '.webp': 'image/webp'
            }
            mime_type = mime_type_map.get(ext, 'image/jpeg')
    except Exception as e:
        print(f"Error reading image file: {e}")
        return

    url = "https://api.anthropic.com/v1/messages"
    headers = {
        "Authorization": f"Bearer {token}",
        "anthropic-beta": "oauth-2025-04-20", # Required for Claude Code OAuth tokens
        "anthropic-version": "2023-06-01",
        "content-type": "application/json"
    }

    # Using the latest Opus 4.6 model found in the binary
    payload = {
        "model": "claude-opus-4-6",
        "max_tokens": 4096,
        "messages": [
            {
                "role": "user",
                "content": [
                    {
                        "type": "image",
                        "source": {
                            "type": "base64",
                            "media_type": mime_type,
                            "data": encoded_string
                        }
                    },
                    {
                        "type": "text",
                        "text": prompt
                    }
                ]
            }
        ]
    }

    print(f"Sending request to Claude Opus 4.6 ({mime_type})...")
    try:
        response = requests.post(url, headers=headers, json=payload)
        response.raise_for_status()
        result = response.json()
        print("\n--- Claude Opus 4.6 Analysis ---\n")
        print(result["content"][0]["text"])
    except requests.exceptions.HTTPError as e:
        print(f"HTTP Error: {e.response.status_code}")
        try:
            error_details = e.response.json()
            print(json.dumps(error_details, indent=2))
        except:
            print(e.response.text)
    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python3 vision_solo.py <image_path> [prompt]")
    else:
        img_path = sys.argv[1]
        user_prompt = sys.argv[2] if len(sys.argv) > 2 else "Analyze this image in detail."
        analyze_image(img_path, user_prompt)

File: vision-solo.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const os = require('os');

const CRED_PATH = path.join(os.homedir(), '.claude', '.credentials.json');

async function get_token() {
    try {
        if (!fs.existsSync(CRED_PATH)) {
            console.error(`Error: Credentials file not found at ${CRED_PATH}`);
            console.error("Please login using 'claude login' first.");
            return null;
        }
        const creds = JSON.parse(fs.readFileSync(CRED_PATH, 'utf8'));
        return creds.claudeAiOauth?.accessToken;
    } catch (e) {
        console.error(`Error reading credentials: ${e.message}`);
        return null;
    }
}

function get_mime_type(filePath) {
    const ext = path.extname(filePath).toLowerCase();
    const map = {
        '.jpg': 'image/jpeg',
        '.jpeg': 'image/jpeg',
        '.png': 'image/png',
        '.gif': 'image/gif',
        '.webp': 'image/webp'
    };
    return map[ext] || 'image/jpeg';
}

async function analyze_image(imagePath, prompt = "Analyze this image in detail.") {
    const token = await get_token();
    if (!token) return;

    if (!fs.existsSync(imagePath)) {
        console.error(`Error: Image file not found at ${imagePath}`);
        return;
    }

    try {
        const imageBuffer = fs.readFileSync(imagePath);
        const base64Image = imageBuffer.toString('base64');
        const mimeType = get_mime_type(imagePath);

        console.log(`\x1b[36mSending request to Claude Opus 4.6 (${mimeType})...\x1b[0m`);

        const response = await fetch("https://api.anthropic.com/v1/messages", {
            method: "POST",
            headers: {
                "Authorization": `Bearer ${token}`,
                "anthropic-beta": "oauth-2025-04-20",
                "anthropic-version": "2023-06-01",
                "Content-Type": "application/json"
            },
            body: JSON.stringify({
                model: "claude-opus-4-6",
                max_tokens: 4096,
                messages: [
                    {
                        role: "user",
                        content: [
                            {
                                type: "image",
                                source: {
                                    type: "base64",
                                    media_type: mimeType,
                                    data: base64Image
                                }
                            },
                            {
                                type: "text",
                                text: prompt
                            }
                        ]
                    }
                ]
            })
        });

        if (!response.ok) {
            const errorText = await response.text();
            console.error(`\x1b[31mAPI Error (${response.status}):\x1b[0m`, errorText);
            return;
        }

        const result = await response.json();
        console.log("\n\x1b[32m--- Claude Opus 4.6 Analysis ---\x1b[0m\n");
        console.log(result.content[0].text);
        console.log("\n\x1b[32m--------------------------------\x1b[0m");

    } catch (e) {
        console.error(`An error occurred: ${e.message}`);
    }
}

// CLI Entry Point
const [,, imgPath, ...promptParts] = process.argv;

if (!imgPath || imgPath === '--help' || imgPath === '-h') {
    console.log(`
\x1b[1mVision Solo\x1b[0m - Standalone Image Analysis via Claude Code Token

\x1b[1mUsage:\x1b[0m
  vision-solo <image_path> [prompt]

\x1b[1mExamples:\x1b[0m
  vision-solo screenshot.png
  vision-solo chart.jpg "What are the trends in this graph?"

\x1b[2mNote: Requires active login in 'claude' CLI.\x1b[0m
    `);
} else {
    const userPrompt = promptParts.length > 0 ? promptParts.join(' ') : "Analyze this image in detail.";
    analyze_image(imgPath, userPrompt);
}
